unlock-app/src/reducers/transactionReducer.js

```diff
@@ -1,28 +1,29 @@
 import { SET_TRANSACTION, UPDATE_TRANSACTION } from '../actions/transaction'
 
-const initialState = null
+const initialState = {
+  latest: null,
+  all: {},
+  lastUpdated: 0,
+}
 
-const transactionReducer = (state = initialState, action) => {
+const transactionReducer = (transactions = initialState, action) => {
+  let newTransactions = Object.assign({}, transactions)
 
   if (action.type === SET_TRANSACTION) {
     if (action.transaction) {
-      return {
-        ...action.transaction,
-      }
+      newTransactions.latest = Object.assign({}, action.transaction)
```

Comment:
Yikes. Is that because React would not refresh if you did not clone?

Reply:
Yep! I checked into it - that's expected behavior. You need to clone at all levels of nesting. Thanks, React.